### PR TITLE
Add lifecycle for controller container to helm chart

### PR DESCRIPTION
**Pull request by coolbry95**
### Proposed changes
Adding lifecycle to controller container. #3006 

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: deployments/helm-chart/templates/controller-daemonset.yaml [hunk 1]
```diff
@@ -71,6 +71,9 @@ spec:
       - name: {{ include "nginx-ingress.name" . }}
         image: "{{ .Values.controller.image.repository }}:{{ .Values.controller.image.tag }}"
         imagePullPolicy: "{{ .Values.controller.image.pullPolicy }}"
+{{- if .Values.controller.lifecycle }}
+{{ toYaml .Values.controller.lifecycle | indent 8 }}
```
Comment (lucacome) [inline deployments/helm-chart/templates/controller-daemonset.yaml:75]:
```suggestion
        lifecycle:
{{ toYaml .Values.controller.lifecycle | indent 10 }}
```

File: deployments/helm-chart/templates/controller-deployment.yaml [hunk 2]
```diff
@@ -76,6 +76,9 @@ spec:
       - image: "{{ .Values.controller.image.repository }}:{{ .Values.controller.image.tag }}"
         name: {{ include "nginx-ingress.name" . }}
         imagePullPolicy: "{{ .Values.controller.image.pullPolicy }}"
+{{- if .Values.controller.lifecycle }}
+{{ toYaml .Values.controller.lifecycle | indent 8 }}
```
Comment (lucacome) [inline deployments/helm-chart/templates/controller-deployment.yaml:80]:
```suggestion
        lifecycle:
{{ toYaml .Values.controller.lifecycle | indent 10 }}
```

**Review by lucacome — CHANGES_REQUESTED** [review]

File: deployments/helm-chart/templates/controller-daemonset.yaml [hunk 3]
```diff
@@ -71,6 +71,10 @@ spec:
       - name: {{ include "nginx-ingress.name" . }}
         image: "{{ .Values.controller.image.repository }}:{{ .Values.controller.image.tag }}"
         imagePullPolicy: "{{ .Values.controller.image.pullPolicy }}"
+{{- if .Values.controller.lifecycle }}
+        lifecycle:
+{{ toYaml .Values.controller.lifecycle | indent 10 }}
+{{- end -}}
```
Comment (lucacome) [inline deployments/helm-chart/templates/controller-daemonset.yaml:77]:
```suggestion
{{- end }}
```

File: deployments/helm-chart/templates/controller-deployment.yaml [hunk 4]
```diff
@@ -76,6 +76,10 @@ spec:
       - image: "{{ .Values.controller.image.repository }}:{{ .Values.controller.image.tag }}"
         name: {{ include "nginx-ingress.name" . }}
         imagePullPolicy: "{{ .Values.controller.image.pullPolicy }}"
+{{- if .Values.controller.lifecycle }}
+        lifecycle:
+{{ toYaml .Values.controller.lifecycle | indent 10 }}
+{{- end -}}
```
Comment (lucacome) [inline deployments/helm-chart/templates/controller-deployment.yaml:82]:
```suggestion
{{- end }}
```

**Review by lucacome — APPROVED** [review]
Small change, but LGTM!

**Review by vepatel — APPROVED** [review]

**Review by jjngx — APPROVED** [review]

**Comment by coolbry95** [comment]
Is there anything left to do here or can we merge this?

**Pull request merged**